### ISSUE TITLE
[Bug] CAT transaction list "To" address is different when re-synced (or new sync)

### ISSUE THREAD

**Issue by pinweihuang** [Chia-Network/chia-blockchain#10913]
### What happened?

As shown below, the `To address` are shown differently in the CAT transaction history of the light-wallet on two different machines.
`xch14xehrr92vcknpukkpu3u9ta8s7g45xrq5mvjwga42dzcumhd9q8sskkk2a` is the desired `To address` but seems like the resulted `To address` is `xch1fqysvc7r5axx7aw77ephzfhtk4v3dp5tvfw7x6x5p02egtme7usq9uzxv4`. And xch scan confirms that `xch1fqysvc7r5axx7aw77ephzfhtk4v3dp5tvfw7x6x5p02egtme7usq9uzxv4` is the one I sent to actually, which is not pointed to the recipient's wallet address `xch14xehrr92vcknpukkpu3u9ta8s7g45xrq5mvjwga42dzcumhd9q8sskkk2a`. The recipient doesn't receive the token.

![image](https://user-images.githubusercontent.com/27796746/160289898-a91b12d2-baa7-4757-89c0-699ee8c9a7d1.png)

### Version

1.3.1

### What platform are you using?

Windows and Mac OS

### What ui mode are you using?

GUI

### Relevant log output

Asset Id: `1a384759d34e3fcf28e53357966675e672898afe13b07cb17c926a448bd35600`

**Comment by wallentx** [comment]
Do you know if the recipient has added the CAT asset ID to their wallet? If they haven't, it won't automatically show up.
When you resyncing the wallet, we currently attempt to make a best guess at the wallet address info that is displayed, which is why you are seeing the difference in address.

**Comment by Quexington** [comment]
This is expected behavior.  A puzzle hash is not only a representation of someone's keys, but also the type of asset being sent there.  We do our best to only show the former, but on a fresh resync, the puzzle hashes are going to come out differently because we don't have the necessary info to transform them.  As a result, you get two different puzzle hashes and only the puzzle hash on the machine that made the transaction will work for sending money.

Basically, don't send to puzzle hashes from your tx list.  We could probably signal this in the GUI a bit clearer.

It is still techincally possible to salavge any money you have sent there but it's a little involved.  Here's a gist made by a community member that is an exerpt of how to create the spend bundle: https://gist.github.com/ojura/332410ddc8de19205249004b2185da24

**Comment by pinweihuang** [comment]
> Do you know if the recipient has added the CAT asset ID to their wallet? If they haven't, it won't automatically show up. When you resyncing the wallet, we currently attempt to make a best guess at the wallet address info that is displayed, which is why you are seeing the difference in address.

The recipient does have added the CAT asset ID to their wallet. Thanks for clarification, now I understand why I am seeing different addresses. However, CAT is still missing. I will follow Quexington's advice below to see if I can rescue the missing CAT I have sent.

**Comment by pinweihuang** [comment]
> Basically, don't send to puzzle hashes from your tx list. We could probably signal this in the GUI a bit clearer.
However, just a few days ago I sent to an old (1.5 month-old) puzzle hashes (_xch15rl3emsn6u4yhdfvlmr8g9srs2nstn5sfprrhmw5p8vhu7psr96stlraet_) generated from `Receive Address` and the same missing sent CAT issue happened again. As shown in the attached screen shot, the green-circled tx did get through and the recipient received the 500 CAT but the red-circled tx didn't get through and the 380 CAT is missing now.

<img width="1194" alt="Screen Shot 2022-04-28 at 8 42 49 AM" src="https://user-images.githubusercontent.com/27796746/165791839-109dde62-e7d7-4f20-8e92-54d98511a23f.png">
